src/expertsystem/reaction/particle.py

```diff
@@ -194,17 +165,38 @@ class Particle:  # pylint: disable=too-many-instance-attributes
         converter=optional(_to_parity), default=None
     )
 
-    @isospin.validator
-    def __check_gellmann_nishijima(self, attribute, value) -> None:  # type: ignore  # pylint: disable=unused-argument
-        if (
-            self.isospin is not None
-            and GellmannNishijima.compute_charge(self) != self.charge
+    def __attrs_post_init__(self) -> None:
+        if not gellmann_nishijima(
```

Comment:
this just needs a `if self.isospin not None and not gellman....(`

Reply:
```suggested
if self.isospin not None and not gellmann_nishijima(
```